src/modules/nunjucks/DateFilter.ts

```diff
@@ -0,0 +1,27 @@
+export function dateFilter(value: string ): string {
+    // tslint:disable-next-line: prefer-const
+    let [year, month, day] = value.split('-').map(i => {
```

Comment:
Shouldn't formatting function be more graceful so that if value is not present it does not blow up? I'd expect this to return undefined or null if input value is undefined or null.

Reply:
I would prefer this to fail loudly than silently, as it is a function which is only used by other programmer-defined functions, rather than as a direct result of user input. If this filter breaks, the person writing the code using it should know about it.

Unless quiet failing is preferred in TypeScript, in which case I'd be happy to be more graceful.

Reply:
I generally prefer to fail fast and loud. I thought maybe allowing undefined / null values would be convenient for developers so no 'if' statements needs to be added in the code but you might be right, I do not find good use case for UI that would print empty string if date value wasn't defined. In that case ignore my initial comment.